packages/cloud_firestore/CHANGELOG.md

```diff
@@ -1,3 +1,9 @@
+## 0.12.2
+
+* Ensure that all channel calls to the Dart side from the Java side are done
+  on the UI thread. This change allows Transactions to work with upcoming
+  Engine restrictions, which require channel calls be made on the UI thread. 
+
```

Comment:
Does this take into account that Firestore's Android SDK will not allow transactions on the UI thread?

Reply:
The transaction itself is not being executed on the UI thread, only the message being passed over the channel from native side to Dart side is being done on the UI thread.